Source/Core/DolphinQt/SkylanderPortal/SkylanderPortalWindow.cpp

```diff
@@ -92,71 +228,367 @@ void SkylanderPortalWindow::CreateMainWindow()
+  m_collection_path = m_path_edit->text();
+  RefreshList();
+}
+
+void SkylanderPortalWindow::SelectPath()
+{
+  QString dir = QDir::toNativeSeparators(DolphinFileDialog::getExistingDirectory(
+      this, tr("Select Skylander Collection"), m_collection_path));
+  if (!dir.isEmpty())
+  {
+    dir += QString::fromStdString("\\");
+    m_path_edit->setText(dir);
+    m_collection_path = dir;
+  }
+  if (m_only_show_collection->isChecked())
+    RefreshList();
+}
+
+void SkylanderPortalWindow::UpdateSelectedVals()
+{
+  const u32 sky_info = m_skylander_list->currentItem()->data(1).toUInt();
+  if (sky_info != 0xFFFFFFFF)
+  {
+    sky_id = sky_info >> 16;
+    sky_var = sky_info & 0xFFFF;
+  }
+}
+
+void SkylanderPortalWindow::RefreshList()
+{
+  int row = m_skylander_list->currentRow();
+  m_skylander_list->clear();
+  if (m_only_show_collection->isChecked())
+  {
+    QDir collection = QDir(m_collection_path);
+    auto& system = Core::System::GetInstance();
+    for (auto file : collection.entryInfoList(QStringList(tr("*.sky"))))
+    {
+      File::IOFile sky_file(file.filePath().toStdString(), "r+b");
+      if (!sky_file)
+      {
+        continue;
+      }
+      std::array<u8, 0x40 * 0x10> file_data;
+      if (!sky_file.ReadBytes(file_data.data(), file_data.size()))
+      {
+        continue;
+      }
+      auto ids = system.GetSkylanderPortal().CalculateIDs(file_data);
+      if (PassesFilter(file.baseName(), ids.first, ids.second))
+      {
+        const uint qvar = (ids.first << 16) | ids.second;
+        QListWidgetItem* skylander = new QListWidgetItem(file.fileName());
+        skylander->setData(1, qvar);
+        m_skylander_list->addItem(skylander);
+      }
+    }
+  }
+  else
+  {
+    for (const auto& entry : IOS::HLE::USB::list_skylanders)
+    {
+      int id = entry.first.first;
+      int var = entry.first.second;
+      if (PassesFilter(tr(entry.second), id, var))
+      {
+        // const QBrush bground = QBrush(QColor(100, 0, 0, 0));
+        const uint qvar = (entry.first.first << 16) | entry.first.second;
+        QListWidgetItem* skylander = new QListWidgetItem(tr(entry.second));
+        QBrush bground = QBrush();
+        if (filters.PassesFilter(SkylanderFilters::G_SPYROS_ADV, id, var))
+          bground = QBrush(QColor(240, 255, 240, 255));
+        else if (filters.PassesFilter(SkylanderFilters::G_GIANTS, id, var))
+          bground = QBrush(QColor(255, 240, 215, 255));
+        else if (filters.PassesFilter(SkylanderFilters::G_SWAP_FORCE, id, var))
+          bground = QBrush(QColor(240, 250, 255, 255));
+        else if (filters.PassesFilter(SkylanderFilters::G_TRAP_TEAM, id, var))
+          bground = QBrush(QColor(255, 240, 240, 255));
+        else if (filters.PassesFilter(SkylanderFilters::G_SUPERCHARGERS, id, var))
+          bground = QBrush(QColor(245, 230, 195, 255));
+        skylander->setBackground(bground);
+        skylander->setData(1, qvar);
+        m_skylander_list->addItem(skylander);
```

Comment:
There probably needs to be a better way to handle this, potentially within the existing themes? As well as this, the text is basically non-existant using a Mac in dark mode